frappe/core/doctype/uniqueness_rule/uniqueness_rule.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from frappe.model.document import Document
+
+class UniquenessRule(Document):
+	pass
+
+def check_uniqueness(doc, method):
+	new_doc = doc.as_dict()
+	doctype_in_uniqueness_rule = frappe.db.get_value("Uniqueness Rule",{"doctype_name":new_doc.doctype})
+	if doctype_in_uniqueness_rule:
+		uniqueness_doc = frappe.get_doc("Uniqueness Rule",{"doctype_name":new_doc.doctype})
```

Comment:
this should be cached too. If there are large number of inserts (data import) you don't need this being queried again and again.